packages/amplify-graphql-index-transformer/src/utils.ts

```diff
@@ -50,3 +51,58 @@ export function validateNotSelfReferencing(config: IndexDirectiveConfiguration |
     }
   }
 }
+
+/**
+ * Checks if @auth owner field has been set to a sortKeyField for @primaryKey
+ */
+export const validateNotOwnerAuth = (
+  sortKeyField: string,
+  config: PrimaryKeyDirectiveConfiguration,
+  ctx: TransformerContextProvider,
+): boolean => {
+  const { object } = config;
+
+  const authDir = object.directives?.find(dir => dir.name.value === 'auth');
+
+  if (!authDir) return true;
+
+  const dirRules = (authDir.arguments?.find(arg => arg.name.value === 'rules')?.value as ListValueNode).values;
+  const ownerRules = dirRules.filter(rule => {
+    let isOwner = false;
+    let hasIdentityClaimField = false;
+    let hasOwnerFieldField = false;
+    let usesMultiClaim = false;
+    let sortKeyFieldIsAuthField = false;
+
+    (rule as ObjectValueNode).fields?.forEach(field => {
+      const name = field.name.value;
+      const { value } = field?.value as StringValueNode;
+
+      if (name === 'allow' && value === 'owner') {
+        isOwner = true;
+      }
+
+      if (name === 'identityClaim') {
+        hasIdentityClaimField = true;
+
+        if (value === 'sub::username') {
+          usesMultiClaim = true;
```

Comment:
nit: If possible, can we replace these checks with a switch-case to reduce complexity.

Reply:
I think a switch case makes this more confusing because sometimes it needs 2 values in the conditional, and sometimes it needs 1.